File: src/main/java/net/cjsah/console/exceptions/Para0CommandException.java
package net.cjsah.console.exceptions;

import net.cjsah.console.command.StringReader;

public class Para0CommandException implements CommandExceptionType{
    private final String message;

    public Para0CommandException(final String message) {
        this.message = message;
    }

    public CommandException create() {
        return new CommandException(this, this.message);
    }

    public CommandException createWithContext(StringReader reader) {
        return new CommandException(this, this.message, reader.getString(), reader.getCursor());
    }

    @Override
    public String toString() {
        return this.message;
    }
}

File: src/main/kotlin/net/cjsah/console/exceptions/Para0CommandException.kt
@file:Suppress("unused")

package net.cjsah.console.exceptions

import net.cjsah.console.command.StringReader

class Para0CommandException(private val message: String) : CommandExceptionType {

    fun create() = CommandException(this, message)

    fun createWithContext(reader: StringReader) = CommandException(this, message, reader.getString(), reader.getCursor())

    override fun toString() = message

}